### DISCUSSION TITLE
Question on external soil model example in Anura3D Tutorial Manual

### DISCUSSION THREAD

**Discussion by JayfansChou** [Anura3D/Anura3D_OpenSource#267] [Q&A]
Dear all,

I’m Chou, and I’m currently learning MPM by following the Anura3D Tutorial Manual. I have a question regarding the external soil model. In Chapter 5, Section 5.5, there is an example of implementing a user-defined constitutive model in Anura3D, which I believe is based on the classical elastic constitutive model. However, I’m a bit confused about the UMAT code, particularly the material parameter "PLIM."

From my understanding, the classical elastic constitutive model typically only requires two parameters to describe the stress-strain behavior, such as Young's Modulus (E) and Poisson’s Ratio (v). However, the Tutorial Manual lists three parameters—E, v, and PLIM—which has left me unsure about the meaning of PLIM. Could you clarify whether this is indeed the classical elastic constitutive model and explain the role of PLIM?

Thank you for your assistance!

![Example UMAT code](https://github.com/user-attachments/assets/4ed9efed-b149-4614-a0bc-bf41628c5cad)

**Comment by Koekphon** [comment]
Have you got the answer yet, I have a bit confused about this as well.

**Reply by AgustinToroF** [reply]
Hi. Did you add the external soil model?
